### PR TITLE
Hdf5 1 12 1

### PR DESCRIPTION
**Pull request by seanm**
## PR Checklist
- [x] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Comment by seanm** [comment]
@dzenanz so this is a WIP update of HDF5... but I need some help to finish... for me, it builds fine as static libs, but not as shared libs, it reports:

```
 CMake Error at CMake/ITKModuleMacros.cmake:395 (set_property):
   set_property could not find TARGET hdf5-static.  Perhaps it has not yet
   been created.
 Call Stack (most recent call first):
   CMake/ITKModuleMacros.cmake:443 (itk_module_target_name)
   Modules/ThirdParty/HDF5/src/CMakeLists.txt:101 (itk_module_target)
```

but I don't know much about cmake, and could not figure our how to fix this...

**Comment by seanm** [comment]
@thewtex @jhlegarreta @N-Dekker maybe any of you have to time/ability to help here...?

**Comment by dzenanz** [comment]
I was planning to take a look on Monday. I know that Matt is busy. I don't know about Jon and Niels' schedules.

**Comment by jhlegarreta** [comment]
Not really my expertise the dance required by this. Really sorry :roll_eyes:.

**Comment by seanm** [comment]
@dzenanz ok cool, was just doing a friendly ping :)

**Comment by dzenanz** [comment]
This week has been busy for me, having been on vacation for two weeks before that 😄

**Comment by dzenanz** [comment]
This build failure might have been due to https://github.com/InsightSoftwareConsortium/ITKPythonPackage/pull/179. I don't see the option to re-run the pipelines.

File: Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h
Comment (dzenanz) [inline Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h:93]: Can you amend this commit to only add symbols? That should also re-trigger the CI.

**Comment by thewtex** [comment]
> This build failure might have been due to InsightSoftwareConsortium/ITKPythonPackage#179. I don't see the option to re-run the pipelines.

This is unrelated.

File: Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h
Comment (seanm) [reply Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h:93]: Not easily I don't think. As you know the file is generated by running the commands at the top of the file.

Why do you ask?  Do you think it has removed some that are in fact needed?

Comment (dzenanz) [reply Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h:93]: It will remove many old symbols, but this will remove some platform-specific symbols for other platforms (e.g. Windows). The risk of somebody using the old HDF5 symbols for something else in their code is minimal.

Just keeping the additions can be done by manual diff viewer manipulation. Do you want to do it yourself @seanm, or do you want me to take over?

Comment (seanm) [reply Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h:93]: Wouldn't it be better for someone to run the symbol mangling on Windows *also*, then merge the results?

Comment (seanm) [reply Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h:93]: After all, there could be new Windows-only symbols in this new HDF that we'll miss otherwise.

Comment (dzenanz) [reply Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h:93]: OK, I will do both (remove deletions and add Windows-specific symbols).

Comment (seanm) [reply Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h:93]: The much bigger problem though is as per my first comment, figuring out why the shared lib build fails...

**Comment by dzenanz** [comment]
This fails to even configure on my computer:
```text
Setting 'CMAKE_INSTALL_RPATH' variable with default value '$ORIGIN/../lib'
nifti_installed_targets: ITKznz;ITKniftiio
Check size of __float128
Check size of __float128 - failed
Check size of _Quad
Check size of _Quad - failed
CMake Error at Modules/ThirdParty/HDF5/src/itkhdf5/config/cmake/ConfigureChecks.cmake:341 (file):
  file failed to open for reading (No such file or directory):

    C:/Dev/ITK-2019/pac_Cconftest.out
Call Stack (most recent call first):
  Modules/ThirdParty/HDF5/src/itkhdf5/CMakeLists.txt:487 (include)


Testing maximum decimal precision for C - 
maximum decimal precision for C var - 0
CMake Error at C:/Program Files/CMake/share/cmake-3.21/Modules/ExternalData.cmake:837 (message):
  Data file referenced by argument

    DATA{C:/Dev/ITK-git/Testing/Data/Baseline/BasicFilters/LaplacianRecursiveGaussianImageFilterTest.png,:}

  corresponds to path

    Testing/Data/Baseline/BasicFilters/LaplacianRecursiveGaussianImageFilterTest.png

  that does not match regular expression

    ()(\.[^./]*)$
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.21/Modules/ExternalData.cmake:708 (_ExternalData_arg_series)
  C:/Program Files/CMake/share/cmake-3.21/Modules/ExternalData.cmake:542 (_ExternalData_arg)
  C:/Program Files/CMake/share/cmake-3.21/Modules/ExternalData.cmake:355 (ExternalData_expand_arguments)
  cmake/ITKModuleTest.cmake:127 (ExternalData_add_test)
  Modules/Filtering/ImageFeature/test/CMakeLists.txt:169 (itk_add_test)


Configuring incomplete, errors occurred!
See also "C:/Dev/ITK-2019/CMakeFiles/CMakeOutput.log".
See also "C:/Dev/ITK-2019/CMakeFiles/CMakeError.log".
```
But it also fails to configure on master:
```text
nifti_installed_targets: ITKznz;ITKniftiio
CMake Error at C:/Program Files/CMake/share/cmake-3.20/Modules/ExternalData.cmake:837 (message):
  Data file referenced by argument

    DATA{C:/Dev/ITK-git/Testing/Data/Baseline/BasicFilters/ConnectedComponentImageFilterTest.png,:}

  corresponds to path

    Testing/Data/Baseline/BasicFilters/ConnectedComponentImageFilterTest.png

  that does not match regular expression

    ()(\.[^./]*)$
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.20/Modules/ExternalData.cmake:708 (_ExternalData_arg_series)
  C:/Program Files/CMake/share/cmake-3.20/Modules/ExternalData.cmake:542 (_ExternalData_arg)
  C:/Program Files/CMake/share/cmake-3.20/Modules/ExternalData.cmake:355 (ExternalData_expand_arguments)
  cmake/ITKModuleTest.cmake:127 (ExternalData_add_test)
  Modules/Segmentation/ConnectedComponents/test/CMakeLists.txt:30 (itk_add_test)


Configuring incomplete, errors occurred!
```
I tried with CMake 3.21.2, 3.21.4, 30.22.0.rc2 and 3.20.6, all produce the same error. This wasn't happening until a few days ago. @bradking do know what is going on? You wrote the offending code in 1082a37e6771430b05d4bb2ba7c0d281649423b5.

**Comment by dzenanz** [comment]
I installed Windows 10's _21H2 preview update_ today. Maybe that has something to do with it?

**Comment by dzenanz** [comment]
Can we somehow use VTK's [update to 1.12.1](https://github.com/Kitware/VTK/commit/2ba55193b8f829b652a742723a38fd3c599f47b0#diff-df2c7f513fcaa185adc66828c95c4a2b7e7c65d2f24b472a305944eedd02f49e)? Their fork is [here](https://gitlab.kitware.com/third-party/hdf5/-/commits/for/vtk).

**Comment by hjmjohnson** [comment]
> Can we somehow use VTK's [update to 1.12.1](https://github.com/Kitware/VTK/commit/2ba55193b8f829b652a742723a38fd3c599f47b0#diff-df2c7f513fcaa185adc66828c95c4a2b7e7c65d2f24b472a305944eedd02f49e)? Their fork is [here](https://gitlab.kitware.com/third-party/hdf5/-/commits/for/vtk).

This would be great.  What do we need to do?

**Comment by seanm** [comment]
@hjmjohnson I think this PR is very close.  On my Mac at least, building as static libs works, all tests pass, etc.  Just not sure what's gone wrong with building as shared libs...

**Comment by hjmjohnson** [comment]
FYI:  The latest version of HDF5 seems to have better explicit support for using HDF5 as a subproject.  See: https://github.com/HDFGroup/hdf5/blob/34ae2775ac67f03f352a340352043e499e90141c/CMakeLists.txt#L49. where the explicit example given is from the Paraview source code.

File: Wrapping/TypedefMacros.cmake
Comment (dzenanz) [inline Wrapping/TypedefMacros.cmake:321]: Possibly reduce duplication by using "Possible values (semicolon separated): ${VALID_WRAP_METHODS}"

Comment (hjmjohnson) [reply Wrapping/TypedefMacros.cmake:321]: Sorry, I don't understand this comment.  The "IN_LIST" item requires a variable on the right-hand side, not a list.

Comment (dzenanz) [reply Wrapping/TypedefMacros.cmake:321]: I suggest we print the content of that variable here, to prevent this message getting out of sync with the list which is actually checked.

Comment (dzenanz) [reply Wrapping/TypedefMacros.cmake:321]: I mean output the content of that variable in this message.

**Comment by hjmjohnson** [comment]
@dzenanz The ghostflow errors require changes prior to the merge.  How does one do that?

**Comment by dzenanz** [comment]
You change the upstream 😄 A recent example of that is #3016.

**Comment by dzenanz** [comment]
When ghostflow warnings can be overridden - it does not have to be green.

**Comment by hjmjohnson** [comment]
@seanm @dzenanz Can this be merged without fixing the ghostflow-check-master?  

I have some more improvements for HDF5, but I don't want to address those less important issues in this PR.  I would love to have a review, and comments or approval.

**Comment by dzenanz** [comment]
> Can this be merged without fixing the ghostflow-check-master?

Of course. It is only a warning, sometimes triggered when updating third-party libraries.

File: Modules/ThirdParty/HDF5/src/itkhdf5/config/cmake/ConfigureChecks.cmake
Comment (dzenanz) [inline Modules/ThirdParty/HDF5/src/itkhdf5/config/cmake/ConfigureChecks.cmake:294]: [HDFGroup/hdf5@`0afdf98` (#1332)](https://github.com/HDFGroup/hdf5/pull/1332/commits/0afdf9821c55fc42025db4827ea94dc150c2e2f8) had 2 lines changed. It the second change forgotten here?

**Review by dzenanz — APPROVED** [review]

**Comment by dzenanz** [comment]
I would like @seanm to take another look before we merge.

File: Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h
Comment (dzenanz) [reply Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h:93]: @hjmjohnson I still think we are safer only adding symbols.

**Comment by seanm** [comment]
It's my first day back to work today... will try to get to this this afternoon though...

File: Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h
Comment (hjmjohnson) [reply Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h:93]: @dzenanz  did you add the windows-specific symbols?

Comment (hjmjohnson) [reply Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h:93]: If so, this could be resolved.

**Comment by seanm** [comment]
@hjmjohnson so do your changes fix building as shared libs?

**Comment by hjmjohnson** [comment]
@seanm   I believe so.  I did the following when doing my internal tests on macs and linux.

```bash
rm -rf * ;  cmake -DITK_WRAP_PYTHON:BOOL=ON -DBUILD_SHARED_LIBS:BOOL=ON ../ITK
```

**Comment by hjmjohnson** [comment]
FYI:  Addressed little comments by @dzenanz, and reviewed and adopted a few more fortifications from the VTK hdf fixes related to the '0' as a variable name problem.

**Comment by hjmjohnson** [comment]
I have some more improvements for HDF5 based on inspiration from the VTK customizations, but they feel like they should be added to a different PR after this one is merged.

**Comment by seanm** [comment]
@hjmjohnson you might also check if they are fixed already upstream in their `develop` branch (that's their `master`)

File: Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h
Comment (dzenanz) [reply Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h:93]: It looks like I didn't. I will take a pass at it later this afternoon.

**Comment by hjmjohnson** [comment]
> @hjmjohnson you might also check if they are fixed already upstream in their `develop` branch (that's their `master`)

They are! Almost exactly like the VTK fixes.  Patch was created by the same VTK contributer.

Thanks for pointing me to this.

File: Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h
Comment (dzenanz) [reply Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h:93]: Done. Perhaps the commits could be re-ordered to make Window additions immediately follow Unix update?

**Comment by dzenanz** [comment]
`Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h:1712:20: error: 'itk_H5LTyytext' undeclared (first use in this function); did you mean 'itk_H5LTyylex'?` is a pretty strange error.

**Comment by hjmjohnson** [comment]
> 

That would have come from an automatically created mangling.

**Comment by hjmjohnson** [comment]
@seanm This PR is now ready for your final review (and hopefully approval).  @dzenanz provided windows name mangling.  @hjmjohnson reviewed both upstream/devel and kitwatware for/vtk branches to incorporate changes that minimize the hdf5 exposure to ITK,  improve cross platform building, and increase cmake compatibility.

**Comment by seanm** [comment]
Thanks so much @hjmjohnson for picking up where I dropped the ball!

What is the git incantation to get the changes you've done here into my own local branch?

File: Modules/ThirdParty/HDF5/src/itkhdf5/CMakeLists.txt
Comment (dzenanz) [inline Modules/ThirdParty/HDF5/src/itkhdf5/CMakeLists.txt:6]: This was removed in the previous commit. Was the introduction intentional?

File: Modules/ThirdParty/HDF5/src/itkhdf5/CMakeLists.txt
Comment (dzenanz) [inline Modules/ThirdParty/HDF5/src/itkhdf5/CMakeLists.txt:1]: I was unaware of this block comment style until just now.

**Comment by dzenanz** [comment]
@seanm Something of the form `git fetch seanm`, where `seanm` is a remote name pointing to `git@github.com:seanm/ITK.git`.

**Comment by hjmjohnson** [comment]
```bash
git checkout -b hdf5_12_1_1
git remote add seanm git@github.com:seanm/ITK.git
git fetch seanm
git reset seanm/hdf5_12_1_1 --hard
```

**Comment by seanm** [comment]
The following seemed to work:

```
git fetch origin
git reset origin/hdf5_1_12_1 --hard
```

Now I'll try some builds and tests...

**Comment by seanm** [comment]
@hjmjohnson I'm getting:

```
make[2]: *** No rule to make target `Modules/ThirdParty/HDF5/src/itkhdf5/src/H5Tinit.c', needed by `Modules/ThirdParty/HDF5/src/itkhdf5/src/CMakeFiles/gen_hdf5-static'.  Stop.
```

File: Modules/ThirdParty/HDF5/src/itkhdf5/CMakeLists.txt
Comment (hjmjohnson) [reply Modules/ThirdParty/HDF5/src/itkhdf5/CMakeLists.txt:1]: I just found it as well.  It makes the code a lot cleaner for minimizing diffs.

**Comment by hjmjohnson** [comment]
> 

@sean 

Is this a clean build?  Can you provide a bit more context about the environment that is failing?

I am unable to replicate that error for any of my builds ([shared/static]/[python]/[mac/ubuntu]).  

Hans

**Comment by seanm** [comment]
> Is this a clean build? Can you provide a bit more context about the environment that is failing?

Yes, very vanilla, on macOS 10.14:

```
mkdir ITK-test-bin
cd ITK-test-bin
cmake ../ITK
make -j16
```

**Comment by seanm** [comment]
Building without `-j16` and with `VERBOSE=1` the last few lines output are:

```
[ 82%] Built target H5detect
/Applications/Xcode.app/Contents/Developer/usr/bin/make  -f Modules/ThirdParty/HDF5/src/itkhdf5/src/CMakeFiles/gen_hdf5-static.dir/build.make Modules/ThirdParty/HDF5/src/itkhdf5/src/CMakeFiles/gen_hdf5-static.dir/depend
cd /Users/sean/external/ITK-test-bin && /Applications/CMake.app/Contents/bin/cmake -E cmake_depends "Unix Makefiles" /Users/sean/external/ITK /Users/sean/external/ITK/Modules/ThirdParty/HDF5/src/itkhdf5/src /Users/sean/external/ITK-test-bin /Users/sean/external/ITK-test-bin/Modules/ThirdParty/HDF5/src/itkhdf5/src /Users/sean/external/ITK-test-bin/Modules/ThirdParty/HDF5/src/itkhdf5/src/CMakeFiles/gen_hdf5-static.dir/DependInfo.cmake --color=
/Applications/Xcode.app/Contents/Developer/usr/bin/make  -f Modules/ThirdParty/HDF5/src/itkhdf5/src/CMakeFiles/gen_hdf5-static.dir/build.make Modules/ThirdParty/HDF5/src/itkhdf5/src/CMakeFiles/gen_hdf5-static.dir/build
make[2]: *** No rule to make target `Modules/ThirdParty/HDF5/src/itkhdf5/src/H5Tinit.c', needed by `Modules/ThirdParty/HDF5/src/itkhdf5/src/CMakeFiles/gen_hdf5-static'.  Stop.
make[1]: *** [Modules/ThirdParty/HDF5/src/itkhdf5/src/CMakeFiles/gen_hdf5-static.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by hjmjohnson** [comment]
Fix for Makefiles seems to break Ninja builds and vice versa.  This H5Tinit.c file is a nightmare.

**Comment by seanm** [comment]
> Fix for Makefiles seems to break Ninja builds and vice versa

Darn, because I can report that your last commit fixed my vanilla makefile build.

**Comment by dzenanz** [comment]
`CMAKE_GENERATOR STREQUAL "Ninja"` could be use to do things differently.

**Comment by seanm** [comment]
OK this now builds as both static and shared for me with makefiles.  Will try more combinations tomorrow.

**Comment by hjmjohnson** [comment]
@seanm @dzenanz FYI:  The H5Tiniit.c generating CMake file now very closely matches the upstream HDF5 develop code in its dependency setup.  The for_vtk branch from the Kitware variant is simpler but does not seem to work with makefiles.

**Review by dzenanz — APPROVED** [review]

**Comment by dzenanz** [comment]
Sean, let us know when you are happy.

**Comment by seanm** [comment]
> The for_vtk branch from the Kitware variant is simpler but does not seem to work with makefiles.

Strange, I build VTK with makefiles all the time.  Maybe my build options never need/build HDF5 tho...

> Sean, let us know when you are happy.

Will do testing today...

**Comment by seanm** [comment]
@hjmjohnson before you started working on this PR, one thing I had tested was building with permutations of the HDF5 options presented in ccmake.  But now it seems they are gone.  I've just double checked against plain master and there I see in ccmake options like:

```
 HDF5_BATCH_H5DETECT             *OFF                                                                                                                                                  
 HDF5_ENABLE_HDFS                *OFF                                                                                                                                                  
 HDF5_ENABLE_MIRROR_VFD          *OFF                                                                                                                                                  
 HDF5_ENABLE_ROS3_VFD            *OFF                                                                                                                                                  
 HDF5_IGNORE_DISABLED_FILE_LOCK  *ON                                                                                                                                                   
 HDF5_USE_FILE_LOCKING           *ON                                                                                                                                                   
```

Did you deliberately or accidentally remove these?

**Comment by dzenanz** [comment]
Deliberately. Take a look at commit titled `COMP: Hard-code HDF5 options to suppress messages`.

**Comment by hjmjohnson** [comment]
It was deliberate.  the for_vtk branch did similar hard-coding of these options.

File: Modules/ThirdParty/HDF5/src/itkhdf5/CMakeLists.txt
Comment (hjmjohnson) [reply Modules/ThirdParty/HDF5/src/itkhdf5/CMakeLists.txt:6]: Minimizing diffs.  Not needed for ITK's minimum cmake version, but is needed for HDF5's minimum version.

I decided to keep it in for now.

**Comment by seanm** [comment]
@hjmjohnson did you try building with python wrapping enabled?  When I try, it seems to freeze here:

image

hmm, but then configuring again freezes in a different place:

image

I'll try on master...

**Comment by seanm** [comment]
Hmm, if I use `cmake` instead of `ccmake` it works. maybe it's a timeout or something.

**Comment by hjmjohnson** [comment]
Did you press "e" to exit the screen?  I think the message being displayed has reached an internal cmake limit.  I don't think it is "froze," I think the message is truncated.

**Comment by seanm** [comment]
`e` works, but then I can only do `c` not `g`. But anyway, I've gotten past it with plain `cmake` and now building with python... 70% done...

**Comment by seanm** [comment]
Boy, building with python sure slows the build!  But I've tried debug, release, and various other permutations of build options and things build and tests pass.  `itkPyBufferMemoryLeakTest` failed once, but not a second time.

So +1 for me!

**Comment by hjmjohnson** [comment]
@seanm @dzenanz THANK YOU.  Nice to have that monkey off my back.  I'm off for vacation next week, and I'm glad this is finished.  Have a great week next week.

**Comment by seanm** [comment]
I should have tried this *before* we merged, but... I've just noticed building my own app, which uses both VTK & ITK, now has link errors:

```
duplicate symbol '_H5C_dump_cache' in:
    /Users/builder/rogue_nightly_build/thirdparty_dependencies/ITK-BS-TSan-install/lib/libitkhdf5-static.a(H5Cdbg.c.o)
    /Users/builder/rogue_nightly_build/thirdparty_dependencies/VTK-BS-TSan-install/lib/libvtkhdf5.a(H5Cdbg.c.o)
duplicate symbol '_H5C_dump_cache_LRU' in:
    /Users/builder/rogue_nightly_build/thirdparty_dependencies/ITK-BS-TSan-install/lib/libitkhdf5-static.a(H5Cdbg.c.o)
    /Users/builder/rogue_nightly_build/thirdparty_dependencies/VTK-BS-TSan-install/lib/libvtkhdf5.a(H5Cdbg.c.o)
duplicate symbol '_H5C_verify_tag' in:
    /Users/builder/rogue_nightly_build/thirdparty_dependencies/ITK-BS-TSan-install/lib/libitkhdf5-static.a(H5Ctag.c.o)
    /Users/builder/rogue_nightly_build/thirdparty_dependencies/VTK-BS-TSan-install/lib/libvtkhdf5.a(H5Ctag.c.o)
```

Very likely due to missing mangling, perhaps in *both* VTK and ITK... I'll work on a fix...

**Comment by dzenanz** [comment]
Mangling update removed a whole lot of symbols. We should probably restore the symbols which were deleted from the mangle list.

**Comment by seanm** [comment]
So these 3 functions exist or not depending on preprocessing:
- the first two are wrapped in `#ifndef NDEBUG`.
- the last one in `#if H5C_DO_TAGGING_SANITY_CHECKS`

I guess to be fully correct, when generating the mangling, we need to do it for release and debug, on different OSes, and with different build options, and merge the results.

For this conflict to have occurred, means that *both* ITK and VTK are failing to mangle these 3.

I'll make patches tomorrow...

**Comment by dzenanz** [comment]
> when generating the mangling, we need to do it for release and debug, on different OSes, and with different build options, and merge the results.

We should put this in the [comments on top of the mangle file](https://github.com/InsightSoftwareConsortium/ITK/blob/v5.3rc02/Modules/ThirdParty/HDF5/src/itkhdf5/src/itk_hdf5_mangle.h#L52-L55).

**Comment by seanm** [comment]
https://github.com/InsightSoftwareConsortium/ITK/pull/3082

https://gitlab.kitware.com/vtk/vtk/-/merge_requests/8813

**Pull request merged**
